src/main/java/teammates/common/util/Url.java

```diff
@@ -107,34 +65,38 @@ public String get(String parameterName) {
      * </ul>
      */
     public static String addParamToUrl(String url, String key, String value) {
-        if (key == null || value == null) {
-            // return the url if any of the key or the value is null
+        if (key == null || key.isEmpty() || value == null || value.isEmpty()
+             || url.contains("?" + key + "=") || url.contains("&" + key + "=")) {
+            // return the url if any of the key or the value is null or empty
+            // or if the key is already included in the url
             return url;
         }
-        if (url.contains("?" + key + "=") || url.contains("&" + key + "=")) {
-            // return the url if the key is already included in the url
-            return url;
-        }
-        url += url.indexOf('?') >= 0 ? '&' : '?';
-        url += key + "=" + Sanitizer.sanitizeForUri(value);
-        return url;
+        return url + (url.contains("?") ? "&" : "?") + key + "=" + Sanitizer.sanitizeForUri(value);
     }
 
     public static String trimTrailingSlash(String url) {
         return url.trim().replaceAll("/(?=$)", "");
     }
+    
+    public static String getRelativePath(String url) {
+        try {
+            return new Url(url).toString();
+        } catch (AssertionError e) {
+            return url;
+        }
+    }
```

Comment:
This code makes me uneasy. 
1. We are not supposed to catch Assertion errors. An assertion error indicates a bug in the code. If you wanted to use the exception to identify malformed URLs, then you should throw the exception from the constructor, catch it here. It is not right to first interpret it as an assumption failure and then interpret it as a malformed url that is sort of expected.
2. I know this issue was raised elsewhere and is planned to be fixed. But to reiterate, the equivalence between toString() and relative path goes against the universal meaning of `toString()`.
3. Try to write header comments for public methods. Usually it is a good test for the API quality. A well-designed method is easy to document.
4. The correct name for the current implementation seems like `getRelativePathIfFullUrl_ReturnSameStringIfNotProperUrl(String url)` which is slightly off from what you actually intended

Consider this:
```java
/** Returns the relative path of the url. Returns an empty string if the url does not have a relative part.
  * @param url : must be a proper URL
  */
public static String getRelativePath(String url) {
   return new Url(url).getRelativePath();
}
```
This code will not puzzle the reader even if the reader knows nothing about the rest of the code, and there will be an assertion error if the code is misused. Not only it works as advertised, it doesn't do anything that is not advertised. The only downside is that it cannot be used if you caller doesn't know whether the url is a full one or a relative one. I think that is a good thing. Methods should be as strict as they can be. Do we have code calling this method with relative URLs without knowing whether the parameter is full or relative?